interfaces/builtin/udisks2.go

```diff
@@ -375,51 +378,69 @@ func (iface *udisks2Interface) Name() string {
 func (iface *udisks2Interface) StaticInfo() interfaces.StaticInfo {
 	return interfaces.StaticInfo{
 		Summary:              udisks2Summary,
+		ImplicitOnClassic:    true,
 		BaseDeclarationSlots: udisks2BaseDeclarationSlots,
 	}
 }
 
 func (iface *udisks2Interface) DBusConnectedPlug(spec *dbus.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	spec.AddSnippet(udisks2ConnectedPlugDBus)
+	if !release.OnClassic {
```

Comment:
I don't understand why the introduction of implicit slot on classic would affect the plug.

This snippet here is exactly:
```
<policy context="default">
    <deny own="org.freedesktop.UDisks2"/>
    <deny send_destination="org.freedesktop.UDisks2"/>
</policy>
```
Reading this snippet though I actually understand this even less.

How come a *connected plug* gets a denial to send to `org.freedesktop.UDisks2`?
Since your patch makes this apply only to core systems it is not a problem on classic but I still don't get the logic behind the original snippet.

Reply:
It is important to remember that the DBus backend is about setting dbus bus policy. On a core system, there is no bus policy for udisks2 so snapd defines it on behalf of the snap. We determined in the initial implementation that on core a reasonable default is for non-root to not have access to udisks2 since there is no polkit integration.

On classic though, udisks2 is shipped as a deb or rpm and so it defines the DBus bus policy, polkit, udev rules, etc, etc, so on classic, snapd should just use those values rather than trying to compete with the deb/rpm installation.

Reply:
The classic part I understand (because there's no snapd-implied policy there which makes perfect sense to me). On core systems though, does it mean that the snippet doesn't prevent root from accessing udisks?

Reply:
Yes. Something needs to be able to use to it. :)

More seriously, the idea is that a) we don't want to allow privilege escalation by the non-root unconfined user,  b) polkit is not available on Core and c) we want snaps to be able to use the interface. We don't care about unconfined root on the device because unconfined root can simply rewrite the dbus policy (or polkit policy, if polkit was on Core).

For 'a', we deny non-root users from accessing the privileged service which is sufficient since a non-root user can't rewrite the dbus bus policy. 'b' is not a concern since non-root isn't allowed any access due to 'a'. For 'c', we use AppArmor to mediate the access instead of DBus bus policy for confined root processes.

*If* we had polkit on Core, we would likely (in addition to AppArmor) use the default bus policy for udisks2 and ship polkit policy to authorize access for the non-root user, but since we don't, we can only reasonably allow root and deny non-root via DBus bus policy.

Reply:
Thank you for explaining this. It all makes sense now.